packages/core/src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -294,13 +291,19 @@ export async function setTransformationToRunningState() {
     await setContextVariables()
     await vscode.commands.executeCommand('aws.amazonq.transformationHub.reviewChanges.reset')
     transformByQState.setToRunning()
-    sessionPlanProgress['startJob'] = StepProgress.Pending
-    sessionPlanProgress['buildCode'] = StepProgress.Pending
-    sessionPlanProgress['generatePlan'] = StepProgress.Pending
-    sessionPlanProgress['transformCode'] = StepProgress.Pending
+    jobPlanProgress['startJob'] = StepProgress.Pending
+    jobPlanProgress['buildCode'] = StepProgress.Pending
+    jobPlanProgress['generatePlan'] = StepProgress.Pending
+    jobPlanProgress['transformCode'] = StepProgress.Pending
     transformByQState.resetPlanSteps()
+    transformByQState.resetSessionJobHistory()
+    transformByQState.setJobId('') // so that details for last job are not overwritten when running one job after another
```

Comment:
The purpose of persisting the last job's ID was not very clear to me, since a back-to-back job should ideally be persisting the context for the new job?

Reply:
The `sessionJobHistory` maps jobIds to details about the job, and in the beginning of a 2nd job, if the previous job's ID is not cleared, the `updateHistory` function running on an interval could put that ID back in the map.